daffodil-runtime1/src/main/scala/org/apache/daffodil/debugger/InteractiveDebugger.scala

```diff
@@ -555,20 +558,15 @@ class InteractiveDebugger(runner: InteractiveDebuggerRunner, eCompilers: Express
         if (args != "") {
           if (dc == cmd) {
             val completeString = getCompleteString(args)
-            val subcandidates = new java.util.ArrayList[CharSequence]
-            val newCursor = subcommandsCompleter.complete(completeString, cursor, subcandidates)
+            val subline = reader.getParser.parse(completeString, 0)
+            val subcandidates = new java.util.ArrayList[Candidate]
+            subcommandsCompleter.complete(reader, subline, subcandidates)
             val seq = subcandidates.asScala
             seq.foreach(c => candidates.add(c))
-            buffer.lastIndexOf(completeString) + newCursor
-          } else {
-            -1
           }
         } else {
           if (dc.name.startsWith(cmd)) {
-            candidates.add(dc.name + " ")
-            buffer.lastIndexOf(cmd)
-          } else {
-            -1
+            candidates.add(new Candidate(dc.name + " "))
```

Comment:
I wonder if adding this space is no longer correct? I'm not sure why it was added originally, but maybe it was to work around some issue with autocomplete that has since been fixed. I've tested removing this, e.g.:

```suggestion
            candidates.add(new Candidate(dc.name))
```
And it does fix the issue with the escaped space.

Reply:
Yes, we should remove that extra space.  You're probably right that the original author of that code added the space to work around some issue with autocomplete that has since been fixed.  I suspect that the subcandidate completion code below is causing the issue you noticed with main command completions sneaking into subcommands too.